File: hw/gavilanez/u3/HW29Strategy/Strategy/src/main/java/ec/edu/espe/strategy/model/BubbleSort.java

package ec.edu.espe.strategy.model;

import ec.edu.espe.strategy.controller.SortingStrategy;

/**
 *
 * @author Kenny Gavilanez, Dev Dynasty, DCCO-ESPE
 */
public class BubbleSort extends SortingStrategy{

    public BubbleSort() {
    }
    @Override
    public void sort(int[] array) {
        for (int reverseIndex = array.length - 1; reverseIndex > 1; reverseIndex--) {

            for (int i = 0; i < reverseIndex; i++) {

                if (array[i] > array[i + 1]) {
                    int temp = array[i];

                    array[i] = array[i + 1];
                    array[i + 1] = temp;

                }
            }
        }

    }
}

File: exams/viteri/U3/FinalExam Viteri/FinalExam Viteri/utils/Bubble_Sort.py
class Bubble_Sort:
    def bubble_sort(list):

        for n in range(len(list)-1, 0, -1):
            for i in range(n):
                if list[i] > list[i + 1]:
                    temp = list[i]
                    list[i] = list[i + 1]
                    list[i + 1] = temp


